backend/service/topology/cache.go

```diff
@@ -43,14 +43,13 @@ func (c *client) acquireTopologyCacheLock(ctx context.Context) {
 		default:
 			c.log.Error("lost connection to database, trying to reconnect...", zap.Error(err))
 		}
+		conn.Close()
```

Comment:
add comment saying it releases lock

Reply:
I actually already have one when I open the conn a 20 lines up or so

Reply:
i can add an extra one here also tho